engine/crates/integration-tests/src/federation/builder/bench.rs

```diff
@@ -152,7 +152,7 @@ impl DummyFetcher {
 
 #[async_trait::async_trait]
 impl runtime::fetch::FetcherInner for DummyFetcher {
-    async fn post(&self, _request: FetchRequest<'_>) -> FetchResult<FetchResponse> {
+    async fn post(&self, _request: &FetchRequest<'_>) -> FetchResult<FetchResponse> {
```

Comment:
We have to take the request by reference for retries, otherwise we'd have to recreate it in the caller.